CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JMultiChoiceComboBoxMenuNGTest.java

```diff
@@ -0,0 +1,369 @@
+        assertEquals(newIcon2, result);
+    }
+
+    /**
+     * Test of getItems method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testGetItems() {
+        System.out.println("getItems");
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+        Set result = instance.getItems();
+        assertEqualsNoOrder(result.toArray(), items.toArray());
+    }
+
+    /**
+     * Test of getSelectedItems method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testGetSelectedItems() {
+        System.out.println("getSelectedItems");
+        String item = "Item 3";
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+        instance.setSelectedItem(item);
+        Set resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 1);
+        assertEquals(resultSet.stream().findFirst().get().toString(), item);
+    }
+
+    /**
+     * Test of setSelectedItem method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testSetSelectedItem() {
+        System.out.println("setSelectedItem");
+        String item = "Item 3";
+        JMultiChoiceComboBoxMenu instance = mock(JMultiChoiceComboBoxMenu.class);
+        instance.setSelectedItem(item);
+        verify(instance, times(1)).setSelectedItem(item);
+    }
+
+    /**
+     * Test of setSelectedItems method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testSetSelectedItems() {
+        System.out.println("setSelectedItems");
+        String s1 = "Item 1";
+        String s2 = "Item 3";
+        String s3 = "Item 4";
+        JMultiChoiceComboBoxMenu instance = mock(JMultiChoiceComboBoxMenu.class);
+        instance.setSelectedItems(s1, s2, s3);
+        verify(instance, times(1)).setSelectedItems(s1, s2, s3);
+    }
+
+    /**
+     * Test of addSelectedItem method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testAddSelectedItem() {
+        System.out.println("addSelectedItem");
+        String item = "Item 3";
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+        instance.addSelectedItem(item);
+        Set<String> resultSet = instance.getSelectedItems();
+        assertEquals(resultSet.size(), 1);
+        assertEquals(resultSet.stream().findFirst().get().toString(), item);
+    }
+
+    /**
+     * Test of addSelectedItems method, of class JMultiChoiceComboBoxMenu.
+     */
+    @Test
+    public void testAddSelectedItems() {
+        System.out.println("addSelectedItems");
+        String s1 = "Item 1";
+        String s2 = "Item 3";
+        String s3 = "Item 4";
+        String[] selectedItems = {s1, s2, s3};
+        JMultiChoiceComboBoxMenu instance = new JMultiChoiceComboBoxMenu("Text", items);
+        instance.addSelectedItems(s1, s2, s3);
+        Set resultSet = instance.getSelectedItems();
```

Comment:
Are you able to add the parameterized type for the set rather than using the raw type?